### PR TITLE
1634: Refactoring out Inline Action Commands and fixing Zoom in the Map Editor

### LINKED ISSUE #1634: Map Editor in 47.3 ignoring 'View' menu 'Zoom' commands.
v47.3, Windows
megameklog.txt file does not appear to state Java version, but I think I'm on the current version.

On the new Map Editor, while [CTRL]+mousewheel when over the main map window will zoom the map, the 'Zoom' commands on the 'View' menu no longer seem to work.

### PR DESCRIPTION
**Pull request by Windchild292**
This fixes #1634, and implements a refactor of action commands in ClientGUI, plus a few other fixes and formatting changes.

This needs to go in with MekHQ PR 1479.

### CODE REVIEW

File: megamek/src/megamek/client/ui/swing/ClientGUI.java [hunk 1]
```diff
@@ -202,7 +248,7 @@
     /**
      * Map each phase to the name of the card for the main display area.
      */
-    private HashMap<String, String> mainNames = new HashMap<String, String>();
+    private HashMap<String, String> mainNames = new HashMap<>();
```
Comment (NickAragua) [inline megamek/src/megamek/client/ui/swing/ClientGUI.java]: Might as well go through and make these Maps if possible.

File: megamek/src/megamek/client/ui/swing/ClientGUI.java [hunk 2]
```diff
@@ -2332,14 +2387,12 @@ public String getDescription() {
...
         JFileChooser fc = new JFileChooser(".");
-        fc
-                .setLocation(frame.getLocation().x + 150,
-                        frame.getLocation().y + 100);
+        fc.setLocation(frame.getLocation().x + 150, frame.getLocation().y + 100);
         fc.setDialogTitle(Messages.getString("BoardEditor.saveAsImage"));
         fc.setFileFilter(new FileFilter() {
             @Override
             public boolean accept(File dir) {
-                return (null != dir.getName()) && (dir.getName().endsWith(".png") || dir.isDirectory()); //$NON-NLS-1$
```
Comment (NickAragua) [inline megamek/src/megamek/client/ui/swing/ClientGUI.java]: Why take out the null check?

File: megamek/src/megamek/client/ui/swing/ClientGUI.java [hunk 3]
```diff
@@ -669,180 +715,197 @@ private void showRoundReport() {
...
-                .equals(VIEW_TOGGLE_FIRING_SOLUTIONS)) {
-            GUIPreferences.getInstance().setFiringSolutions(
-                    !GUIPreferences.getInstance().getFiringSolutions());
-            if (!GUIPreferences.getInstance().getFiringSolutions()) {
-                bv.clearFiringSolutionData();
-            } else {
-                if (curPanel instanceof FiringDisplay) {
-                    ((FiringDisplay) curPanel).setFiringSolutions();
+        switch (event.getActionCommand()) {
+            case FILE_GAME_SAVE:
```
Comment (NickAragua) [inline megamek/src/megamek/client/ui/swing/ClientGUI.java]: I'll assume you kept the internal logic the same and just changed the if/else block to a switch statement.

File: megamek/src/megamek/client/ui/swing/ClientGUI.java (hunk 2 above)
Comment (Windchild292) [reply megamek/src/megamek/client/ui/swing/ClientGUI.java]: It is unnecessary, as pointed out by my IDE

File: megamek/src/megamek/client/ui/swing/ClientGUI.java (hunk 3 above)
Comment (Windchild292) [reply megamek/src/megamek/client/ui/swing/ClientGUI.java]: Exactly

File: megamek/src/megamek/client/ui/swing/ClientGUI.java (hunk 1 above)
Comment (Windchild292) [reply megamek/src/megamek/client/ui/swing/ClientGUI.java]: Fixed

**Comment by NickAragua** [comment]
Fix #1634

**Pull request merged**